src/rebar_relx.erl

```diff
@@ -195,6 +199,32 @@ merge_overlays(Config) ->
     NewOverlay = lists:flatmap(fun({overlay, Overlay}) -> Overlay end, lists:reverse(Overlays)),
     [{overlay, NewOverlay} | Others].
 
+overlay_vars(Opts) ->
+    case proplists:get_value(overlay_vars, Opts) of
+        undefined ->
+            [];
+        OverlayVars ->
+            [OverlayVars]
```

Comment:
is this right? The name `OverlayVars` implies the result is a list, which you put in another list. The callsite goes `[{base_dir, rebar_dir:base_dir(State)} | overlay_vars(Opts)]` which implies that if you have overlay vars `[{a, b}, {c, d}]` (or even just `[a, b]`), this function will return `[[{a,b}, {c,d}]]` (or `[[a,b]]`) and will always get a bad result at the callsite.

If what you get back isn't a list, then the variable name is at least misleading.

Reply:
I see. That was wrong. I've fixed that to return the `overlay_vars` which should point to 'Path to a file of overlay variables'.